native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2ConnectionManager.java

```diff
@@ -195,8 +200,7 @@ public void closeChannelAndEvict(Http2ClientChannel http2ClientChannel) {
                 if (!result) {
                     logger.warn("Specified channel does not exist in the stale list.");
                 }
-                http2ClientChannel.getConnection()
-                        .close(new DefaultPromise(new DefaultEventLoop()));
+                http2ClientChannel.getConnection().close(new DefaultPromise(new DefaultEventLoop()));
```

Comment:
To be on the safe side, can we pass a `ChannelPromise` using `http2ClientChannel.getChannel().newPromise()`?
```suggestion
                http2ClientChannel.getConnection().close(http2ClientChannel.getChannel().newPromise());
```

Reply:
Updated